### PR TITLE
chore: Use only new tracker fields name in JdbcEventStore [TECH-1526]

### PR DESCRIPTION
**Pull request by enricocolasante**
Following up on https://github.com/dhis2/dhis2-core/pull/13500

* Review naming of fields and just use new tracker names for event fields

NOTE: we are still in the 🚧 phase going with broad strokes through web/service/repo layers. We will come back to all these layers and have discussions on how we want to adapt them. First goal is for new tracker to become independent from dxf2. All code that maps from `dxf2` to `dhis-api` will be removed very soon 👷🏻 Please comment if you spot a bug but don't comment on its style 💅🏻 

# Next

* then tracked entity
* remove dxf2 dependencies in new tracker relationship service
* make sure we have a new tracker store for every entity and do not need to rely on dxf2 or dhis-api stores

### CODE REVIEW

File: dhis-2/dhis-services/dhis-service-tracker/src/main/java/org/hisp/dhis/tracker/event/DefaultEventService.java [hunk 1]
```diff
@@ -278,9 +278,9 @@ public void validate( EventSearchParams params, User user )
         }
 
         if ( violation == null && params.hasLastUpdatedDuration()
```
Comment (teleivo) [inline dhis-2/dhis-services/dhis-service-tracker/src/main/java/org/hisp/dhis/tracker/event/DefaultEventService.java:280]:
```suggestion
        if ( violation == null && params.hasUpdatedAtDuration()
```

File: dhis-2/dhis-services/dhis-service-tracker/src/main/java/org/hisp/dhis/tracker/event/EventSearchParams.java [hunk 2]
```diff
@@ -61,23 +61,23 @@
 
     public static final String EVENT_ENROLLMENT_ID = "enrollment";
 
-    public static final String EVENT_CREATED_ID = "created";
+    public static final String EVENT_CREATED_AT_ID = "createdAt";
 
-    public static final String EVENT_CREATED_BY_USER_INFO_ID = "createdbyuserinfo";
+    public static final String EVENT_CREATED_BY_ID = "createdby";
```
Comment (teleivo) [inline dhis-2/dhis-services/dhis-service-tracker/src/main/java/org/hisp/dhis/tracker/event/EventSearchParams.java:66]:
```suggestion
    public static final String EVENT_CREATED_BY_ID = "createdBy";
```

File: dhis-2/dhis-services/dhis-service-tracker/src/main/java/org/hisp/dhis/tracker/event/EventSearchParams.java [hunk 3]
```diff
@@ -265,23 +265,23 @@ public boolean hasProgramStage()
      */
     public boolean hasLastUpdatedStartDate()
```
Comment (teleivo) [inline dhis-2/dhis-services/dhis-service-tracker/src/main/java/org/hisp/dhis/tracker/event/EventSearchParams.java:266]:
```suggestion
    public boolean hasUpdatedAtStartDate()
```

File: dhis-2/dhis-services/dhis-service-tracker/src/main/java/org/hisp/dhis/tracker/event/EventSearchParams.java [hunk 4]
```diff
@@ -265,23 +265,23 @@ public boolean hasProgramStage()
...
     {
-        return lastUpdatedStartDate != null;
+        return updatedAtStartDate != null;
     }
 
     /**
      * Indicates whether this parameters specifies a last updated end date.
      */
     public boolean hasLastUpdatedEndDate()
```
Comment (teleivo) [inline dhis-2/dhis-services/dhis-service-tracker/src/main/java/org/hisp/dhis/tracker/event/EventSearchParams.java:274]:
```suggestion
    public boolean hasUpdatedAtEndDate()
```

File: dhis-2/dhis-services/dhis-service-tracker/src/main/java/org/hisp/dhis/tracker/event/EventSearchParams.java [hunk 5]
```diff
@@ -265,23 +265,23 @@ public boolean hasProgramStage()
...
     {
-        return lastUpdatedEndDate != null;
+        return updatedAtEndDate != null;
     }
 
     /**
      * Indicates whether this parameters has a lastUpdatedDuration filter.
      */
     public boolean hasLastUpdatedDuration()
```
Comment (teleivo) [inline dhis-2/dhis-services/dhis-service-tracker/src/main/java/org/hisp/dhis/tracker/event/EventSearchParams.java:282]:
```suggestion
    public boolean hasUpdatedAtDuration()
```

File: dhis-2/dhis-services/dhis-service-tracker/src/main/java/org/hisp/dhis/tracker/event/EventSearchParams.java [hunk 6]
```diff
@@ -467,58 +467,58 @@ public EventSearchParams setEventStatus( EventStatus eventStatus )
...
-        this.lastUpdatedDuration = lastUpdatedDuration;
+        this.updatedAtDuration = updatedAtDuration;
         return this;
     }
 
-    public Date getDueDateStart()
+    public Date getScheduleAtStart()
     {
-        return dueDateStart;
+        return scheduleAtStart;
```
Comment (teleivo) [inline dhis-2/dhis-services/dhis-service-tracker/src/main/java/org/hisp/dhis/tracker/event/EventSearchParams.java:505]: Should we make this consistent with the `updatedAt` ones which are all `updatedAtStartDate/EndDate`?

So either `scheduleAtStartDate` or remove the `Date` suffix from the other fields/setters/getters

File: dhis-2/dhis-services/dhis-service-tracker/src/main/java/org/hisp/dhis/tracker/event/JdbcEventStore.java [hunk 7]
```diff
@@ -193,19 +193,18 @@ public class JdbcEventStore implements EventStore
         builder.put( EVENT_ORG_UNIT_ID, "ou_uid" );
         builder.put( EVENT_ORG_UNIT_NAME, "ou_name" );
         builder.put( "trackedEntityInstance", "tei_uid" );
```
Comment (teleivo) [inline dhis-2/dhis-services/dhis-service-tracker/src/main/java/org/hisp/dhis/tracker/event/JdbcEventStore.java:195]: is now `trackedEntity`

File: dhis-2/dhis-services/dhis-service-tracker/src/main/java/org/hisp/dhis/tracker/event/EventSearchParams.java [hunk 8]
```diff
@@ -61,23 +61,23 @@
...
+    public static final String EVENT_CREATED_AT_ID = "createdAt";
 
-    public static final String EVENT_CREATED_BY_USER_INFO_ID = "createdbyuserinfo";
+    public static final String EVENT_CREATED_BY_ID = "createdby";
 
-    public static final String EVENT_LAST_UPDATED_ID = "lastUpdated";
+    public static final String EVENT_UPDATED_AT_ID = "updatedAt";
 
-    public static final String EVENT_LAST_UPDATED_BY_USER_INFO_ID = "lastUpdatedbyuserinfo";
+    public static final String EVENT_LAST_UPDATED_ID = "updatedBy";
```
Comment (teleivo) [inline dhis-2/dhis-services/dhis-service-tracker/src/main/java/org/hisp/dhis/tracker/event/EventSearchParams.java:70]:
```suggestion
    public static final String EVENT_UPDATED_BY_ID = "updatedBy";
```

File: dhis-2/dhis-e2e-test/src/test/java/org/hisp/dhis/tracker/importer/TrackerExportTests.java [hunk 9]
```diff
@@ -383,6 +384,17 @@ public void shouldReturnFilteredEvent()
             .body( "event", equalTo( "ZwwuwNp6gVd" ) );
     }
 
+    @Test
+    void shouldReturnInvalidPropertyWhenOrderOnLegacyCreatedField()
```
Comment (teleivo) [inline dhis-2/dhis-e2e-test/src/test/java/org/hisp/dhis/tracker/importer/TrackerExportTests.java:388]: Can you please add

`// TODO(tracker): remove with old tracker`

so we remember to remove it one day 😅

**Review by teleivo — CHANGES_REQUESTED** [review]
Yeah 😄 we are getting closer 🥳  

I think `EventSearchParams.EVENT_CREATED_BY_ID` has a casing issue (which is why I hit the request changes).

Apart from that just some minor things we can do in this PR or another one. I went through https://docs.dhis2.org/en/develop/using-the-api/dhis-core-version-master/new-tracker.html#changes-in-the-api
and it looks good to me

**Review by teleivo — APPROVED** [review]

**Review by muilpp — APPROVED** [review]

**Pull request merged**